docs/user-guide/databases.rst

```diff
@@ -0,0 +1,208 @@
+#############
+SQL Databases
+#############
+
+Felis can create a database from a schema using the command line tool or the Python API.
+This will emit DDL which can either be executed automatically or written to a file for later use.
+
+Using the Command Line Tool
+===========================
+
+The ``felis create`` command can be used to create a database from the command line.
+For more information on the command line options for the ``create`` command, see the
+`command line documentation <cli.html#felis-create>`_.
+
+Dry Run Mode
+------------
+
+The dry run mode can be used to generate SQL without actually creating the database.
+This can be useful for inspecting the SQL that will be generated before running it.
+
+There are several different ways to run in dry run mode, one being to simply omit the ``--engine-url`` option:
+
+.. code-block:: bash
+
+    felis create schema.yaml
+
+This will by default instantiate the database to an in-memory (transient) SQLite database.
+Running such a command can be useful for finding errors in the schema file before attempting to create a
+persistent database.
+
+A specific database dialect such as MySQL can be selected in dry run mode using the ``--engine-url`` option:
+
+.. code-block:: bash
+
+    felis create --engine-url mysql:// schema.yaml
+
+This will generate the SQL for a MySQL database but will not actually create it.
+
+The dry run mode may also be explicitly enabled using the ``--dry-run`` option:
+
+.. code-block:: bash
+
+    felis create --dry-run --engine-url mysql+mysqlconnector://username:password@localhost schema.yaml schema.yaml
+
+The URL in this case will be used to determine the database dialect and generate the appropriate SQL but will
+otherwise be ignored.
```

Comment:
Maybe, to reduce confusion, dry run should only require dialect name instead of full URL?

Reply:
I see your point but changing this is beyond the scope of this PR, since it only deals with documentation.

Being able to execute `felis create --engine-url mysql://` and have dry run mode be enabled automatically is essentially a convenience feature.

Reply:
> dry run should only require dialect name instead of full URL?

That is the case currently. You can use `mysql://` as the engine URL, for instance.

This text is just indicating that if you provide a full URL in dry run mode, everything but the dialect will be ignored by SQLAlchemy (due to how mock engine works).